### PR TITLE
config: remove ApiTypeOracle assert

### PR DESCRIPTION
**Pull request by rebello95**
This assert currently validates that the return value is non-null. However, call sites actually [guard against `nullptr`](https://github.com/envoyproxy/envoy/pull/9618/files#diff-17967d56376bf9821edb17b67b8a7e63R228-R230) and treat it as a valid codepath.

In Envoy Mobile, we're hitting this assert because we use the v3 configuration and don't include all the v2 protos when building the library. Removing the v2 fallback protos in this scenario can be a valid optimization.

Given the above example and the fact that callers explicitly guard against `nullptr`, this assert should be removed and this codepath considered valid.

Risk Level: Low
Testing: Built Envoy / Envoy Mobile locally + CI
Docs Changes: None
Release Notes: None

Signed-off-by: Michael Rebello <me@michaelrebello.com>

### CODE REVIEW

**Comment by rebello95** [comment]
CI failure seems unrelated

File: source/common/config/api_type_oracle.cc [hunk 1]
```diff
@@ -1,8 +1,5 @@
 #include "common/config/api_type_oracle.h"
 
-#include "common/common/assert.h"
-#include "common/common/logger.h"
```
Comment (mattklein123) [inline source/common/config/api_type_oracle.cc]: Do you mind cleaning up the BUILD file also? https://github.com/envoyproxy/envoy/blob/d1db56424d39f10715f81d17d80079a22040abbf/source/common/config/BUILD#L16-L17

Comment (rebello95) [reply source/common/config/api_type_oracle.cc]: good call, updated

Comment (rebello95) [reply source/common/config/api_type_oracle.cc]: assert is still used here, but removed logger: 

https://github.com/envoyproxy/envoy/blob/d1db56424d39f10715f81d17d80079a22040abbf/source/common/config/version_converter.cc#L5

Comment (rebello95) [reply source/common/config/api_type_oracle.cc]: looks like `version_converter_lib` was transitively getting `assert_lib` by depending on `api_type_oracle_lib` even though it uses `assert_lib` directly. fixed the dependencies in the BUILD file accordingly

**Pull request merged**